### PR TITLE
HTTP Performance tests in CI

### PR DESCRIPTION
**Pull request by achamayou**
Add HTTP performance tests in the CI for CFT.

BFT + HTTP does not handle signature verification at the moment, so the tests cannot get past the startup phase. They'll be enabled when that's fixed.

### CODE REVIEW

**Comment by cimetrics** [comment]
http_perf@[3515 aka 20191217.12](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3515&view=results) vs master ewma over 20 builds from [3311](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3311&view=results) to [3513](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3513&view=results)
images

File: .azure-pipelines-templates/matrix.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ parameters:
     CIMetrics:
       container: nosgx
       pool: Ubuntu-1804-D8s_v3
-      dependsOn: ['BuildAndTest_SGX_CFT_FTCP_Perf', 'BuildAndTest_SGX_BFT_FTCP_Perf']
+      dependsOn: ['BuildAndTest_SGX_CFT_FTCP_Perf', 'BuildAndTest_SGX_BFT_FTCP_Perf', 'BuildAndTest_SGX_CFT_HTTP_Perf', 'BuildAndTest_SGX_BFT_HTTP_Perf']
```
Comment (achamayou) [inline .azure-pipelines-templates/matrix.yml]: I'd really like to find a way to generate that from the perf test generation loop, but I really haven't found any way. Suggestions welcome.

**Comment by cimetrics** [comment]
http_perf@[3538 aka 20191218.12](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3538&view=results) vs master ewma over 20 builds from [3318](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3318&view=results) to [3516](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3516&view=results)
images

**Comment by cimetrics** [comment]
http_perf@[3540 aka 20191218.13](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3540&view=results) vs master ewma over 20 builds from [3318](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3318&view=results) to [3516](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3516&view=results)
images

File: .azure-pipelines-templates/common.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
       - template: coverage.yml
         parameters:
           consensus: ${{ parameters.consensus }}
-    - ${{ if eq(parameters.suffix, 'Perf') }}:
+    - ${{ if and(eq(parameters.suffix, 'Perf'), not(and(eq(parameters.consensus, 'BFT'), eq(parameters.protocol, 'HTTP')))) }}:
```
Comment (achamayou) [inline .azure-pipelines-templates/common.yml]: As stateed in PR description.

**Comment by cimetrics** [comment]
http_perf@[3544 aka 20191218.15](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3544&view=results) vs master ewma over 20 builds from [3318](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3318&view=results) to [3516](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3516&view=results)
images

**Pull request merged**